### PR TITLE
Moving ClusterStatsMonitoringDoc to opaque string versions

### PR DESCRIPTION
**Pull request by ldematte**
After conversation with the Kibana team, we discovered that:
- The `cluster_stats` documents produced by this class are used for internal collection
- Kibana only (used to) use it as a source of telemetry data - not anymore
- Cluster stats monitoring is “owned” by stack monitoring
- Stack monitoring is OK with removing it, but thinks it might be good to still have it as opaque metadata for the on-prem users 
> stack monitoring server code has guards where the version is retrieved and is already expecting the property to not exist so removing it would not result in breakage. the version is however used in the UI and looks like useful metadata for on prem users so I’d be inclined to keep the value and use something else for serverless (the build hash sounds good)

So instead of removing it, we are migrating it to use `Build.current().version()` and to use String versions for nodes etc.

There are remaining transitive usages in ClusterStatsMonitoringDoc when it writes its XContent:
- PluginDescriptor - this is being dealt with
- DiscoveryNode.VersionInfo (via ClusterState)
- NodeInfo (via ClusterStatsResponse/ClusterStatsNodes - but we already moved this to String as soon as we read it)

These will be handled separately

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsIT.java [hunk 1]
```diff
@@ -208,7 +208,8 @@ public void testValuesSmokeScreen() throws IOException, ExecutionException, Inte
         assertThat(msg, response.nodesStats.getJvm().getVersions().size(), Matchers.greaterThan(0));
 
         assertThat(msg, response.nodesStats.getVersions().size(), Matchers.greaterThan(0));
-        assertThat(msg, response.nodesStats.getVersions().contains(Version.CURRENT), Matchers.equalTo(true));
+        // TODO: Build.current().unqualifiedVersion() -- or Build.current().version() if/when we move NodeInfo to Build version(s)
+        assertThat(msg, response.nodesStats.getVersions().contains(Version.CURRENT.toString()), Matchers.equalTo(true));
```
Comment (thecoop) [inline server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsIT.java:212]: `getVersions()` is a collection, so you can replace this with

```suggestion
        assertThat(msg, response.nodesStats.getVersions(), hasItem(Version.CURRENT.toString()));
```

File: server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java [hunk 2]
```diff
@@ -137,8 +136,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
 
         builder.startArray(Fields.VERSIONS);
-        for (Version v : versions) {
-            builder.value(v.toString());
+        for (final var v : versions) {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java:139]: Don't think the `final` adds anything here...

Comment (ldematte) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java:139]: It does not hurt either :) but I can remove it

**Review by williamrandolph — APPROVED** [review]
LGTM

**Pull request merged**